nixos/modules/services/x11/desktop-managers/gnome.nix

```diff
@@ -215,7 +225,9 @@ in
       services.gnome.core-shell.enable = true;
       services.gnome.core-utilities.enable = mkDefault true;
 
-      services.displayManager.sessionPackages = [ pkgs.gnome-session.sessions ];
+      services.displayManager.sessionPackages = [
+        (pkgs.gnome-session.override { x11Support = !cfg.waylandSessionOnly; }).sessions
```

Comment:
The package is also used on another two lines below [L322](https://github.com/getchoo-contrib/nixpkgs/blob/b18b9492f0f0f6bbf3a0f5ea9dd79aece6d78836/nixos/modules/services/x11/desktop-managers/gnome.nix#L322) and [L365](https://github.com/getchoo-contrib/nixpkgs/blob/b18b9492f0f0f6bbf3a0f5ea9dd79aece6d78836/nixos/modules/services/x11/desktop-managers/gnome.nix#L365).

It seems likely that we could end up building/installing multiple versions of the package if we override one but not the others.

Should we bind the overridden package in a let-in block, and replace _all_ references to `pkgs.gnome-session` with it?